x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/search_after_bulk_create.ts

```diff
@@ -136,6 +137,9 @@ export const searchAfterAndBulkCreate = async ({
         // skip the call to bulk create and proceed to the next search_after,
         // if there is a sort id to continue the search_after with.
         if (includedEvents.length !== 0) {
+          if (toReturn.createdSignalsCount + includedEvents.length >= tuple.maxSignals) {
+            toReturn.warningMessages.push(getMaxSignalsWarning(tuple.maxSignals));
+          }
           // make sure we are not going to create more signals than maxSignals allows
           const limitedEvents = includedEvents.slice(
             0,
```

Comment:
Fix this bug too where we truncate the events before passing them in to deduplication logic in `bulkCreate`